pynest/nest/server/hl_api_server.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+#
+# hl_api_server.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import array
+import importlib
+import inspect
+import io
+import sys
+
+import nest
+
+import flask
+from flask import Flask, request, jsonify
+from flask_cors import CORS, cross_origin
+
+from werkzeug.exceptions import abort
+from werkzeug.wrappers import Response
+
+
+__all__ = [
+    'app'
+]
+
+# Blacklist of modules according to Bandit (https://bandit.readthedocs.io).
+_blacklist_modules = [
+    'commands',
+    'dsa',
+    'jinja2',
+    'mako',
+    'os',
+    'paramiko',
+    'popen2',
+    'requests',
+    'rsa',
+    'socket',
+    'ssl',
+    'subprocess',
+    'sys',
+]
+
+_blacklist_builtins = [
+    'copyright',
+    'eval',
+    'exec',
+    'input',
+    'quit',
+    'open',
+]
+
+
+app = Flask(__name__)
+CORS(app)
+
+
+@app.route('/exec', methods=['GET', 'POST'])
```

Comment:
Sorry for putting my two cents in, but

I want to stress the obvious thing: python `exec` function is unsafe.
Sebastian presented this code during today's meeting about NEST Desktop - Elephant integration and we agree on the `/api` endpoint only.
There is a number of tools ([RestrictedPython](https://github.com/zopefoundation/RestrictedPython), [safe_exec](https://github.com/edx/codejail/blob/master/codejail/safe_exec.py)) that aim to sanitize the `exec` payloads, but none of them provides a secure solution. This [comment](https://www.reddit.com/r/Python/comments/6b3zzy/sanitizing_code_for_an_exec_command/) puts it simply:
>  It's completely possible to still do evil things (including importing arbitrary modules) without ever writing the phrase "import". It's possible to do this even if you've tried to restrict the builtins.

Having that said, I don't want to interrupt the merge process of this great PR, because I find it useful and great in all other senses save for `exec()`.

Danylo.

Reply:
:+1: and thanks to @babsey for the recent commits fixing this.